dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileTinyVectorIndexReader.cpp

```diff
@@ -36,9 +36,10 @@ void ColumnFileTinyVectorIndexReader::read(
     vec_column->reserve(read_rows);
     std::vector<Float32> value;
     size_t current_rowid = rowid_start_offset;
-    for (const auto & [rowid, _] : read_rowids)
+    for (const auto & rowid : read_rowids)
     {
-        vec_index->get(rowid, value);
+        // Each ColomnFileTiny has its own vector index, rowid_start_offset is the offset of the ColmnFilePersistSet.
+        vec_index->get(rowid - rowid_start_offset, value);
```

Comment:
These lines fix the recall issue: https://github.com/pingcap/tiflash/issues/9599